### PR TITLE
Autocompletion

### PR DESCRIPTION
**Pull request by akhal3d96**

### CODE REVIEW

File: exe/dirtree [hunk 1]
```diff
@@ -16,7 +22,14 @@ options = {
 }
 
 OptionParser.new do |opts|
-  opts.banner = 'Usage: dirtree [options]... [file]...'
+  opts.banner = <<-BANNER
+  Usage: dirtree [options]... [file]...
+
+  To get autocompletion functionalit
```
Comment (emad-elsaid) [inline exe/dirtree]: `functionality` :)

File: exe/dirtree-completion.bash [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# To get autocompletion functionalit
```
Comment (emad-elsaid) [inline exe/dirtree-completion.bash]: `functionality` also

**Review by emad-elsaid — COMMENTED** [review]
I'm still not sure how it will work when the user execute `dirtree completion` that will execute `dirtree` passing `completion` as the first argument correct? how is that going to output the `dirtree-completion.bash` file?

File: exe/dirtree (hunk 1 above)
Comment (akhal3d96) [reply exe/dirtree]: 😭

**Comment by akhal3d96** [comment]
`$ dirtree completion` prints `dirtree-copletion.bash` which can be added as a file to `/usr/local/etc/bash_completion.d/dirtree` or to be appended to `.bashrc` file and that will enable the auto-completion feature in bash

Resources:
https://github.com/npm/npm/blob/master/lib/utils/completion.sh
https://iridakos.com/tutorials/2018/03/01/bash-programmable-completion-tutorial.html
https://www.tldp.org/LDP/abs/html/tabexpansion.html

**Comment by emad-elsaid** [comment]
Yes but how “dirtree completion” prints “dirtree-completion.bash” ?

Emad Elsaid 
Software Engineer

**Comment by akhal3d96** [comment]
`dirtree-completion.bash` is located at `exe` directory.

**Comment by emad-elsaid** [comment]
Yes and? Why invoking “dirtree completion” prints it? That should execute “dirtree” not “dirtree-completion”

Emad Elsaid 
Software Engineer

**Comment by akhal3d96** [comment]
Executing `dirtree completion` is supposed to print the bash completion script which are located at `exe/dirtree-completion.bash`. then it's up to the user whether he wants append this script to his local bash instance at `.bashrc` or globally `/usr/local/etc/bash_completion.d/dirtree`.

**Comment by emad-elsaid** [comment]
Have you tried that command?

Emad Elsaid 
Software Engineer

**Comment by akhal3d96** [comment]
Yes of course!

**Pull request merged**